apis/core.oam.dev/common/types.go

```diff
@@ -337,95 +312,26 @@ type PolicyStatus struct {
 	Status *runtime.RawExtension `json:"status,omitempty"`
 }
 
-// WorkflowStep defines how to execute a workflow step.
-type WorkflowStep struct {
-	// Name is the unique name of the workflow step.
-	Name string `json:"name"`
-
-	Type string `json:"type"`
-
-	Meta *WorkflowStepMeta `json:"meta,omitempty"`
-
-	// +kubebuilder:pruning:PreserveUnknownFields
-	Properties *runtime.RawExtension `json:"properties,omitempty"`
-
-	SubSteps []WorkflowSubStep `json:"subSteps,omitempty"`
-
-	If string `json:"if,omitempty"`
-
-	Timeout string `json:"timeout,omitempty"`
-
-	DependsOn []string `json:"dependsOn,omitempty"`
-
-	Inputs StepInputs `json:"inputs,omitempty"`
-
-	Outputs StepOutputs `json:"outputs,omitempty"`
-}
-
-// WorkflowStepMeta contains the meta data of a workflow step
-type WorkflowStepMeta struct {
-	Alias string `json:"alias,omitempty"`
-}
-
-// WorkflowSubStep defines how to execute a workflow subStep.
-type WorkflowSubStep struct {
-	// Name is the unique name of the workflow step.
-	Name string `json:"name"`
-
-	Type string `json:"type"`
-
-	Meta *WorkflowStepMeta `json:"meta,omitempty"`
-
-	// +kubebuilder:pruning:PreserveUnknownFields
-	Properties *runtime.RawExtension `json:"properties,omitempty"`
-
-	If string `json:"if,omitempty"`
-
-	Timeout string `json:"timeout,omitempty"`
-
-	DependsOn []string `json:"dependsOn,omitempty"`
-
-	Inputs StepInputs `json:"inputs,omitempty"`
-
-	Outputs StepOutputs `json:"outputs,omitempty"`
-}
-
 // WorkflowStatus record the status of workflow
 type WorkflowStatus struct {
-	AppRevision string       `json:"appRevision,omitempty"`
-	Mode        WorkflowMode `json:"mode"`
-	Message     string       `json:"message,omitempty"`
+	AppRevision string                            `json:"appRevision,omitempty"`
+	Mode        string                            `json:"mode"`
+	Phase       workflowv1alpha1.WorkflowRunPhase `json:"status"`
```

Comment:
a new field you call it `phase` in Go while call it `status` in json?

Reply:
That's what we do in application status...